### PR TITLE
fix: multiple minor errors for Dash Core pre v22

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented

It fixes several minor issues:

 - error messages in logs on regtest:

       [httpworker.0] [masternode/payments.cpp:116] [IsTransactionValid] CMNPaymentsProcessor::IsTransactionValid -- ERROR! Failed to get payees for block at height 110
       [httpworker.0] [masternode/payments.cpp:278] [IsBlockPayeeValid] CMNPaymentsProcessor::IsBlockPayeeValid -- Valid masternode payment at height 110: CTransaction(hash=837e35fab5, ver=3, type=5, vin.size=1, vout.size=1, nLockTime=0, vExtraPayload.size=70)
              CTxIn(COutPoint(0000000000000000000000000000000000000000000000000000000000000000, 4294967295), coinbase 016e0101)
              CTxOut(nValue=500.00000000, scriptPubKey=76a9148708dff2bf8b31363cb4201d)

 - error messages for non-existing files for first run:

        [      init] [util/system.h:57] [error] ERROR: CoreRead: Failed to open file DASH/node0/regtest/governance.dat

 - non-dashified `bitcoin-cli` in `doc/JSON-RPC-interface.md`
 - incorrect initialization of `CMerkleBlock` in functional tests which can cause an un-explainable failure
 - use random ports for `interface_zmq_dash.py` to let tests run simultaneously
 - fix missing todo about using `sethdseed` after #6017
 - optimization of local run of functional tests (slowest go first)
## What was done?
See each commit

## How Has This Been Tested?
Run unit/functional tests

## Breaking Changes
N/A

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
utACK 2026c59d3cafa163a9b954a8abbdb69f841637f5

File: src/flat-database.h [hunk 1]
```diff
@@ -200,10 +200,10 @@ class CFlatDB
         int64_t nStart = GetTimeMillis();
 
         LogPrintf("Writing info to %s...\n", strFilename);
-        CoreWrite(objToSave);
+        const bool ret = CoreWrite(objToSave);
         LogPrintf("%s dump finished  %dms\n", strFilename, GetTimeMillis() - nStart);
```
Comment (kwvg) [inline src/flat-database.h:204]: We should probably not print "dump finished" if `ret` is false

File: src/flat-database.h [hunk 2]
```diff
@@ -200,10 +200,10 @@ class CFlatDB
         int64_t nStart = GetTimeMillis();
 
         LogPrintf("Writing info to %s...\n", strFilename);
-        CoreWrite(objToSave);
+        const bool ret = CoreWrite(objToSave);
```
Comment (kwvg) [inline src/flat-database.h:203]: We should probably not print "dump finished" if `ret` is false

**Review by kwvg — COMMENTED** [review]
LGTM, small suggestion

File: src/flat-database.h (hunk 2 above)
Comment (knst) [reply src/flat-database.h:203]: that log is for benchmark purposes, but I am ready to apply suggestions

File: src/masternode/payments.cpp [hunk 3]
```diff
@@ -51,8 +51,12 @@ CAmount PlatformShare(const CAmount reward)
         LogPrint(BCLog::MNPAYMENTS, "CMNPaymentsProcessor::%s -- MN reward %lld reallocated to credit pool\n", __func__, platformReward);
         voutMasternodePaymentsRet.emplace_back(platformReward, CScript() << OP_RETURN);
     }
-
-    auto dmnPayee = m_dmnman.GetListForBlock(pindexPrev).GetMNPayee(pindexPrev);
+    const auto mnList = m_dmnman.GetListForBlock(pindexPrev);
```
Comment (PastaPastaPasta) [inline src/masternode/payments.cpp:54]: 678db6f43ec9a440a155c0b52cea0287b861aaef: where'd you find this crash?

**Review by PastaPastaPasta — COMMENTED** [review]
generally LGTM 2026c59d3cafa163a9b954a8abbdb69f841637f5; one question

**Review by PastaPastaPasta — APPROVED** [review]
utACK https://github.com/dashpay/dash/commit/2026c59d3cafa163a9b954a8abbdb69f841637f5

**Pull request merged**